### PR TITLE
update RHCOS CI for 4.14

### PR DESCRIPTION
**Pull request by mike-nguyen**

### CODE REVIEW

File: ci-operator/config/coreos/coreos-assembler/coreos-coreos-assembler-rhcos-4.14.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+build_root:
+  project_image:
+    dockerfile_literal: |
+      FROM registry.fedoraproject.org/fedora:37
```
Comment (travier) [inline ci-operator/config/coreos/coreos-assembler/coreos-coreos-assembler-rhcos-4.14.yaml:4]:
```suggestion
      FROM registry.fedoraproject.org/fedora:38
```
?

Comment (travier) [reply ci-operator/config/coreos/coreos-assembler/coreos-coreos-assembler-rhcos-4.14.yaml:4]: Wondering if we missed other instances of that for 4.13

Comment (travier) [reply ci-operator/config/coreos/coreos-assembler/coreos-coreos-assembler-rhcos-4.14.yaml:4]: 4.13 is on F37: https://github.com/coreos/coreos-assembler/blob/rhcos-4.13/Dockerfile

So it's only here

**Review by dustymabe — APPROVED** [review]

File: ci-operator/config/openshift/os/openshift-os-release-4.14.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ base_images:
   coreos_coreos-assembler_latest:
```
Comment (marmijo) [inline ci-operator/config/openshift/os/openshift-os-release-4.14.yaml:2]:
```suggestion
  coreos_coreos-assembler_4.14:
```

In the referenced PR: https://github.com/openshift/release/pull/33905, this line was changed to indicate the version. Want to do that here too?

Comment (mike-nguyen) [reply ci-operator/config/openshift/os/openshift-os-release-4.14.yaml]: I missed that change!  Thanks @marmijo

**Comment by mike-nguyen** [comment]
This job adds a mirror for a container image that is used in the job so pj-rehearse jobs are failing.

**Review by marmijo — APPROVED** [review]

**Pull request merged**